Sources/CustomerCenter/Events/CustomerCenterEvent.swift

```diff
@@ -13,70 +13,127 @@
 
 import Foundation
 
-/// An event to be sent by the `RevenueCatUI` SDK.
-public enum CustomerCenterEvent: FeatureEvent {
+/// A protocol that represents a customer center event.
+public protocol CustomerCenterEventType {}
+
+extension CustomerCenterEventType {
 
-    // swiftlint:disable type_name
+    var feature: Feature { .customerCenter }
 
-    /// An identifier that represents a customer center event.
-    public typealias ID = UUID
+}
 
-    // swiftlint:enable type_name
+/// Data that represents a customer center event creation.
+public struct CustomerCenterEventCreationData {
 
-    /// An identifier that represents a paywall session.
-    public typealias SessionID = UUID
+    let id: UUID
+    let date: Date
 
-    var feature: Feature {
-        return .customerCenter
+    // swiftlint:disable:next missing_docs
+    public init(
+        id: UUID = .init(),
+        date: Date = .init()
+    ) {
+        self.id = id
+        self.date = date
     }
 
+}
+
+/// An event to be sent by the `RevenueCatUI` SDK.
+public enum CustomerCenterEvent: FeatureEvent, CustomerCenterEventType {
+
+    var eventDiscriminator: String? { "lifecycle" }
```

Comment:
Hmm I'm confused by this default event discriminator... Should this be something like `customer_center`? We could also make it a computed property based on the enum values I guess.